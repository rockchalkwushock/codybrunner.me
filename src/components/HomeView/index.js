import React from 'react'

import {
  FlexContainer,
  Grid,
  Heading,
  Icon,
  Link,
  List,
  Text
} from '../commons'
import Card from '../Card'
import ProfilePic from '../ProfilePic'
import SEO from '../SEO'

const ExtendedGrid = Grid.extend`
  grid-template-columns: 3fr;

  > * {
    grid-column: 1 / 4;
  }
`

const Heading4 = Heading.withComponent('h4')

/**
 * @fileOverview
 * Home Page View
 *
 * Accepts siteMetadata & 5 most recent posts from GraphQL Query.
 */

const HomeView = ({ meta, posts }) => (
  <FlexContainer>
    <SEO site={meta} />
    <Grid>
      <ProfilePic />
      <Heading>{meta.author}</Heading>
      <Heading4>{meta.jobTitle}</Heading4>
      <List flow="column" gap={10}>
        {meta.contacts.map(c => (
          <Link ext href={c.href} key={c.label} label={c.label}>
            <Icon className={c.className} />
          </Link>
        ))}
      </List>
    </Grid>
    <ExtendedGrid alternate>
      <Heading>About</Heading>
      <Text>{meta.aboutSnippet}</Text>
      <Text>{meta.employment}</Text>
      <List flow="column" gap={50}>
        {meta.business.map(b => (
          <Link ext href={b.href} key={b.label} label={b.label}>
            <Icon className={b.className} />
          </Link>
        ))}
      </List>
    </ExtendedGrid>
    <Grid alternate>
      <Heading>Recent Posts</Heading>
      <List gap={20}>
        {posts.map(({ node: post }) => (
          <Card key={post.fields.slug} post={post} />
        ))}
      </List>
    </Grid>
  </FlexContainer>
)

export default HomeView
