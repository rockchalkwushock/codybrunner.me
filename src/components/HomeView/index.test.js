import React from 'react'
import { shallow } from 'enzyme'

import HomeView from './'

const { siteMetadata } = require('../../../gatsby-config')

const posts = [
  {
    node: {
      excerpt: 'blah blah blah',
      fields: {
        slug: '/turdsspace/'
      },
      frontmatter: {
        date: 'February 03, 2018',
        draft: false,
        tags: ['JavaScript', 'fp', 'oop'],
        title: 'All About JS'
      }
    }
  }
]

describe('Component: <HomeView />', () => {
  let wrapper
  beforeEach(() => {
    wrapper = shallow(<HomeView meta={siteMetadata} posts={posts} />)
  })

  test('should render correctly', () => {
    expect(wrapper).toMatchSnapshot()
  })
})
