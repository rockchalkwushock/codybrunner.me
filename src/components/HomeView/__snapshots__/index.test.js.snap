// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component: <HomeView /> should render correctly 1`] = `
<FlexContainer>
  <SEO
    site={
      Object {
        "aboutSnippet": "Cody Brunner is a full-stack JavaScript developer & Navy Veteran residing in Wichita, Kansas. Cody primarily works with Node, React, & GraphQL. When not writing code he loves to go hiking, play with his dog & niece, and watchhis Jayhawks win!",
        "author": "Cody Brunner",
        "business": Array [
          Object {
            "className": "fas, envelope",
            "href": "mailto:rockchalkwushock@icloud.com",
            "label": "Email",
          },
          Object {
            "className": "fab, telegram-plane",
            "href": "https://t.me/rockchalkwushock",
            "label": "Telegram",
          },
          Object {
            "className": "fas, file-pdf",
            "href": "https://www.dropbox.com/s/hokjljqc8iob7xd/Cody%20A%20Brunner%20-%20Web%20Developer%20Resume.pdf?dl=1",
            "label": "Resume",
          },
        ],
        "contacts": Array [
          Object {
            "className": "fab, github",
            "href": "https://github.com/rockchalkwushock",
            "label": "Github",
          },
          Object {
            "className": "fab, instagram",
            "href": "https://www.instagram.com/rockchalkwushock",
            "label": "Instagram",
          },
          Object {
            "className": "fab, linkedin",
            "href": "https://www.linkedin.com/in/cody-brunner",
            "label": "LinkedIn",
          },
          Object {
            "className": "fas, rss",
            "href": "rss.xml",
            "label": "RSS Feed",
          },
          Object {
            "className": "fab, twitter",
            "href": "https://twitter.com/RockChalkDev",
            "label": "Twitter",
          },
          Object {
            "className": "fab, youtube",
            "href": "https://www.youtube.com/channel/UCZgBTMhX7jZTkbm7Fpv2bWw",
            "label": "Youtube",
          },
        ],
        "copyright": "Â© 2017-2018 Cody Brunner",
        "description": "Cody Brunner is a Full-Stack JavaScript Developer living in Wichita, Kansas",
        "disqusShortname": "codybrunner-rocks",
        "employment": "Are you interested in contacting Cody for employment opportunities? Feel free to reach out to him via email, Telegram, or go ahead and checkout his resume hosted on DropBox.",
        "googleVerify": "TODO",
        "jobTitle": "Full-Stack JavaScript Developer",
        "keywords": "Cody Brunner, Full-Stack JavaScript Developer, Wichita, web-development, tech blog, Node.js, React.js, Apollo/GraphQL",
        "lang": "en_US",
        "links": Object {
          "creativeCommons": Object {
            "href": "https://creativecommons.org/licenses/by-nc-sa/4.0/",
            "text": "All Rights Reserved.",
          },
          "gatsby": Object {
            "href": "https://gatsbyjs.org",
            "text": "Powered by Gatsby 1.9",
          },
          "now": Object {
            "href": "https://zeit.co/now",
            "text": "Hosted on Now",
          },
          "src": Object {
            "href": "https://github.com/rockchalkwushock/codybrunner.rocks",
          },
          "styled": Object {
            "href": "https://www.styled-components.com",
            "text": "Styled with Styled-Components 3.1",
          },
        },
        "menu": Array [
          Object {
            "href": "/",
            "id": 1,
            "text": "Home",
          },
          Object {
            "href": "/about",
            "id": 2,
            "text": "About",
          },
          Object {
            "href": "/posts/1",
            "id": 3,
            "text": "Blog",
          },
        ],
        "siteUrl": "/",
        "title": "Cody Brunner - Full-Stack JavaScript Developer",
        "twitter": "@RockChalkDev",
      }
    }
  />
  <Grid
    headCard={true}
  >
    <ProfilePic />
    <Heading>
      Cody Brunner
    </Heading>
    <styled.h4>
      Full-Stack JavaScript Developer
    </styled.h4>
    <List
      flow="column"
      gap={10}
    >
      <SiteLink
        ext={true}
        href="https://github.com/rockchalkwushock"
        key="Github"
        label="Github"
      >
        <SiteIcon
          icon={
            Array [
              "fab",
              "github",
            ]
          }
          size="2x"
        />
      </SiteLink>
      <SiteLink
        ext={true}
        href="https://www.instagram.com/rockchalkwushock"
        key="Instagram"
        label="Instagram"
      >
        <SiteIcon
          icon={
            Array [
              "fab",
              "instagram",
            ]
          }
          size="2x"
        />
      </SiteLink>
      <SiteLink
        ext={true}
        href="https://www.linkedin.com/in/cody-brunner"
        key="LinkedIn"
        label="LinkedIn"
      >
        <SiteIcon
          icon={
            Array [
              "fab",
              "linkedin",
            ]
          }
          size="2x"
        />
      </SiteLink>
      <SiteLink
        ext={true}
        href="rss.xml"
        key="RSS Feed"
        label="RSS Feed"
      >
        <SiteIcon
          icon={
            Array [
              "fas",
              "rss",
            ]
          }
          size="2x"
        />
      </SiteLink>
      <SiteLink
        ext={true}
        href="https://twitter.com/RockChalkDev"
        key="Twitter"
        label="Twitter"
      >
        <SiteIcon
          icon={
            Array [
              "fab",
              "twitter",
            ]
          }
          size="2x"
        />
      </SiteLink>
      <SiteLink
        ext={true}
        href="https://www.youtube.com/channel/UCZgBTMhX7jZTkbm7Fpv2bWw"
        key="Youtube"
        label="Youtube"
      >
        <SiteIcon
          icon={
            Array [
              "fab",
              "youtube",
            ]
          }
          size="2x"
        />
      </SiteLink>
    </List>
  </Grid>
  <styled.div
    alternate={true}
  >
    <Heading>
      About
    </Heading>
    <Text>
      Cody Brunner is a full-stack JavaScript developer & Navy Veteran residing in Wichita, Kansas. Cody primarily works with Node, React, & GraphQL. When not writing code he loves to go hiking, play with his dog & niece, and watchhis Jayhawks win!
    </Text>
    <Text>
      Are you interested in contacting Cody for employment opportunities? Feel free to reach out to him via email, Telegram, or go ahead and checkout his resume hosted on DropBox.
    </Text>
    <List
      flow="column"
      gap={50}
    >
      <SiteLink
        ext={true}
        href="mailto:rockchalkwushock@icloud.com"
        key="Email"
        label="Email"
      >
        <SiteIcon
          icon={
            Array [
              "fas",
              "envelope",
            ]
          }
          size="2x"
        />
      </SiteLink>
      <SiteLink
        ext={true}
        href="https://t.me/rockchalkwushock"
        key="Telegram"
        label="Telegram"
      >
        <SiteIcon
          icon={
            Array [
              "fab",
              "telegram-plane",
            ]
          }
          size="2x"
        />
      </SiteLink>
      <SiteLink
        ext={true}
        href="https://www.dropbox.com/s/hokjljqc8iob7xd/Cody%20A%20Brunner%20-%20Web%20Developer%20Resume.pdf?dl=1"
        key="Resume"
        label="Resume"
      >
        <SiteIcon
          icon={
            Array [
              "fas",
              "file-pdf",
            ]
          }
          size="2x"
        />
      </SiteLink>
    </List>
  </styled.div>
  <Grid
    alternate={true}
  >
    <Heading>
      Recent Posts
    </Heading>
    <List
      flow="row"
      gap={20}
    >
      <SiteCard
        key="/turdsspace/"
        post={
          Object {
            "excerpt": "blah blah blah",
            "fields": Object {
              "slug": "/turdsspace/",
            },
            "frontmatter": Object {
              "date": "February 03, 2018",
              "draft": false,
              "tags": Array [
                "JavaScript",
                "fp",
                "oop",
              ],
              "title": "All About JS",
            },
          }
        }
      />
    </List>
  </Grid>
</FlexContainer>
`;
